pkg/cmd/util/cmd.go

```diff
@@ -121,6 +133,18 @@ func convertItemsForDisplay(objs []runtime.Object, preferredVersions ...schema.G
 			}
 		}
 
+		// if no preferred version found in the list of given GroupVersions,
+		// attempt to convert to first GroupVersion that satisfies a preferred version
+		if len(actualOutputVersion.Version) == 0 {
+			for _, version := range preferredVersions {
```

Comment:
You shouldn't even need to do this loop - you should be passing preferredVersions in to ConvertToVersion as a Versioner.